affirm_example/__init__.py

```diff
@@ -16,17 +16,27 @@ def _get_extra_request_args():
         return {}
 
 
-def get_checkout_from_token(checkout_token):
-    read_checkout_url = "{0}/checkout/{1}".format(app.config["AFFIRM"]["API_URL"], checkout_token)
+def _get_secret_api_key(public_api_key):
+    if "MERCHANTS" not in app.config["AFFIRM"]:
+        return app.config["AFFIRM"]["SECRET_API_KEY"]
+
+    for merchant in app.config["AFFIRM"]["MERCHANTS"]:
+        if merchant['PUBLIC_API_KEY'] == public_api_key:
+            return merchant['SECRET_API_KEY']
+
+
+def get_checkout_from_token(checkout_token, public_api_key):
+    read_checkout_url = "{0}/checkout/{1}".format(
+        app.config["AFFIRM"]["API_URL"], checkout_token)
     print read_checkout_url
     return requests.get(read_checkout_url,
                         headers={"Content-Type": "application/json"},
-                        auth=(app.config["AFFIRM"]["PUBLIC_API_KEY"],
-                              app.config["AFFIRM"]["SECRET_API_KEY"]),
+                        auth=(public_api_key,
```

Comment:
It seems there is a chance that `public_api_key` won't match `_get_secret_api_key(public_api_key)` ?

Now I'm thinking do we want to have a function to return a tuple of public key and secret key.
```python
def get_auth_token(public_api_key):
  merchants = app.config["AFFIRM"]["MERCHANTS"]
  if merchants:
      merchant = next((m for m in merchants if m["PUBLIC_API_KEY"] == public_api_key), None)
      if merchant:
        return (merchant["PUBLIC_API_KEY"], merchant["SECRET_API_KEY"])
  return (app.config["AFFIRM"]["PUBLIC_API_KEY"], app.config["AFFIRM"]["SECRET_API_KEY"])
```

So in this case, we don't have to make sure public_api_key is always available and valid - so no need to add `url_config.update(merchant_info)` as well.

wdyt?

Reply:
updated so that existing functionality is 100% the same (in terms of URLs / query params) unless they've toggled a merchant - [e78c24a](https://github.com/Affirm/affirm-example-python/pull/32/commits/e78c24ab93af5f685544e450daccdcdad1378a3a)